### ISSUE TITLE
"keyword_search" failed when the target is None

### ISSUE THREAD

**Issue by xiangce** [RedHatInsights/insights-core#2851]
The following `TypeError` is reported when parsing some insights archives

```
TypeError | condition_error | argument of type 'NoneType' is not iterable | __init__.py, line 514, in <lambda>
```

Traceback:
```
  File "plugin_xx.py", line 62, in is_ha_node
    if ps.search(COMMAND_NAME__contains='pcsd'):
  File "insights-core/insights/combiners/ps.py", line 191, in search
    return keyword_search(self._pid_data.values(), **kwargs)
  File "insights-core/insights/parsers/__init__.py", line 536, in keyword_search
    if all(map(lambda kv: key_match(row, kv[0], kv[1]), kwargs.items())):
  File "insights-core/insights/parsers/__init__.py", line 536, in <lambda>
    if all(map(lambda kv: key_match(row, kv[0], kv[1]), kwargs.items())):
  File "insights-core/insights/parsers/__init__.py", line 532, in key_match
    return key in my_row and matcher_fn(my_row[key], value)
  File "insights-core/insights/parsers/__init__.py", line 514, in <lambda>
    'contains': lambda s, v: v in s,
TypeError: argument of type 'NoneType' is not iterable
```

**Comment by xiangce** [comment]
The root cause of this issue is that the `keyword_search` applies the `match function` to the items with `None` value in the searching target.

E.g.,There is such parsed `ps` result:
```
{'PID': 1077020, 'USER': 'root', 'UID': None, 'PPID': 1076486, '%CPU': 0.0, '%MEM': None, 'VSZ': None, 'RSS': None, 'TTY': '?', 'STAT': None, 'START': '03:23', 'TIME': '00:
00:00', 'COMMAND': 'grep -F CMD crmd openshift start node sap bash chronyd nova-conductor clvmd nova-compute auditd pacemaker-controld neutron-ns-metadata-proxy nginx: worker process /u
sr/bin/openshift start node heat-engine spausedd tuned openshift start master controllers ntpd redis-server dlm_controld /usr/bin/gnome-shell nginx: master process pcsd docker ora coros
ync pacemakerd /usr/bin/openshift start master mongdb openshift start master api', 'COMMAND_NAME': None, 'ARGS': None, 'F': None, 'PRI': None, 'NI': None, 'WCHAN': None, '_line': 'root
    1077020 1076486  0 03:23 ?        00:00:00 grep -F CMD crmd openshift start node sap bash chronyd nova-conductor clvmd nova-compute auditd pacemaker-controld neutron-ns-metadata-pro
xy nginx: worker process /usr/bin/openshift start node heat-engine spausedd tuned openshift start master controllers ntpd redis-server dlm_controld /usr/bin/gnome-shell nginx: master pr
ocess pcsd docker ora corosync pacemakerd /usr/bin/openshift start master mongdb openshift start master api'}
```
When applying "ps.search(COMMAND_NAME__contains='xyz'), the above `TypeError` will be thrown.

And, Why is there such a `ps` result with a `None` `COMMAND_NAME`? It's caused by #2827, which skipped the `'grep -F'` line but kept its pre-parsed result in the `self.data`

**Comment by xiangce** [comment]
#2852 fixes the issue of `keyword_search`
#2853 fixes the issue of `ps`

**Comment by xiangce** [comment]
resolved by #2852 and #2853
